src/report.rs

```diff
@@ -44,7 +46,50 @@ impl Report {
       format!("unix:{:?}", platform.os_type).into()
     };
 
-    let backtrace = format!("{:#?}", Backtrace::new());
+    //We skip non-user code frames, including Backtrace::new()
+    const SKIP_FRAMES_NUM: usize = 8;
+    //Code is based on backtrace source
+    const HEX_WIDTH: usize = mem::size_of::<usize>() * 2 + 2;
+
+    let mut backtrace = String::new();
+
+    for (idx, frame) in Backtrace::new()
+      .frames()
+      .iter()
+      .skip(SKIP_FRAMES_NUM)
+      .enumerate()
+    {
+      let ip = frame.ip();
+      let _ = write!(backtrace, "\n{:4}: {:2$?}", idx, ip, HEX_WIDTH);
```

Comment:
More of a question but: what happens if `write!` fails? I guess there's an `Err(...)` that we just never check?

While we don't _ever_ want to panic, it might be a nicer approach to move this whole code to a function that can return a `Result<...>`, then just using `?` liberally to make sure that no badly formatted stacktraces are written.

Reply:
There is no reason for it to fail for `String`